cirq-core/cirq/protocols/json_serialization.py

```diff
@@ -386,6 +387,21 @@ def default(self, o):
                 'index': o.index,
             }
 
+        # datetime
+        if isinstance(o, datetime.datetime):
+            if o.tzinfo is None or o.tzinfo.utcoffset(o) is None:
+                # Otherwise, the deserialized object may change depending on local timezone.
+                raise TypeError(
+                    "Can only serialize 'aware' datetime objects with `tzinfo`. "
+                    "Consider using e.g. `datetime.datetime.now(tz=datetime.timezone.utc)`"
+                )
+
+            if o.tzinfo != datetime.timezone.utc:
+                # Saves us the trouble of having to serialize the timezone.
+                raise TypeError("Can only serialize UTC timestamps.")
```

Comment:
so if you had

```python
ts = now().in_my_local_timezone()
assert ts == read_json(to_json(ts))
```

the assertion would fail, even though you're still referring to the correct point in time.

Reply:
While round trip equality is nice, I think datetime is generally broken with how it handles naive vs tz-aware instances and so we shouldn't expect roundtrip equality (even two tz-aware datetime instances with the same timestamp in different timezones do not compare equal). So I suggest we support serializing any datetime by getting its `timestamp`, and then we just have to pick whether to deserialize as naive or tz-aware datetimes. I think tz-aware with utc is ok, though I would probably prefer naive for simplicity.

Reply:
ah. Thanks for investigating @wcourtney . It looks like equality _does_ work for "aware" datetimes.

Reply:
( I removed the utc check during serialization)